### DISCUSSION TITLE
When is the Strimzi operator going to graduate from sandbox status?

### DISCUSSION THREAD

**Discussion by jslusher** [strimzi/strimzi-kafka-operator#8322] [General]
We used the Strimzi cluster operator to run a Kafka cluster for a short time a couple of years ago, and we eventually migrated over to a MSK cluster because we weren't sure about Strimzi's stability as a project at the time. After a couple of years using MSK, we're not overly impressed with what it offers. Aside from being expensive, there's no AWS-native way to deal with ACLs, and MSK Connectors are just terrible and terribly overpriced. We will most likely at least use the Strimzi operator to manage our connectors going forward, and we're flirting with the idea of using the Strimzi operator to replace our MSK clusters in production, but what we always come back to is the project's CNCF status of "sandbox". Is that ever going to change? I see that there are a handful of other companies who use it in production. I'm curious to what extent they're using it. Anyways, thanks for the great work on a great operator and best of luck!

**Comment by scholzj** [comment]
First of all, there is a long way to graduation. First, we need to incubate. We proposed the incubation last year (@ppatierno opened the PR) as we believed we fulfil all the criteria. That process is now in progress. But when / if it happens, does not depend entirely on us. The process needs time (as everyone is quite busy, there are many projects trying to incubate / graduate etc.) to complete and of course, it also requires the CNCF TOC to _agree_ that we fit the criteria - they make the decision. So I do not think this is a question we can answer with some exact date. TBH, I'm not sure if the incubation or even graduation gives you any real guarantees and is something you should wait for - but I guess that is up to you.

**Reply by jslusher** [reply]
I saw the [proposal](https://github.com/cncf/toc/pull/842), and as you said it seems to be dependent on a lot of factors out of your control. I agree that graduating doesn't give any actual guarantees about the stability of the product's resources, but it does seem to mean that we can rely on the stability of the project itself. It should be a testament to the project that it's been around as long as it has without a change in its CNCF status. I'm mostly curious about your own expectations and what it would mean to you to graduate. Again thanks for all your hard work!

**Reply by scholzj** [reply]
I don't know, and I guess I might have too much emotional interest to be able to give an objective view. We clearly did mistakes ... at some point we for example decided to do 1.0 release only after ZooKeeper is removed from Kafka since that will bring many changes. And we wanted to propose the incubation with the 1.0 release. But that is clearly taking way longer than we expected at the start. So looking back, both decisions were bad (and we clearly decided to propose the incubation without it, but the ZooKeeper removal now seems so close to not making any sense to do 1.0 now) and might not make Strimzi look like a very mature project. The incubation and graduation criteria are also not an exact formula -> so different projects might have different _confidence_ to go for it or just better social skills. So there are things like that which impact it as well.

On the other hand, there are several companies selling products or services based on Strimzi. I think that means a lot as well and shows a lot of trust.

**Reply by ppatierno** [reply]
Nothing to add to what Jakub already said. Moving to the next step is not on us unluckily even if we are confident that Strimzi meets the needed criteria. It's a really long process.
Imho the important thing is that, as you mentioned, different companies are already using it as part of their product.
You can see the entire list in the [ADOPTERS](https://github.com/strimzi/strimzi-kafka-operator/blob/main/ADOPTERS.md) file.
